applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/geomechanics_wrapper.py

```diff
@@ -0,0 +1,39 @@
+# Importing the Kratos Library
+import KratosMultiphysics as KM
+from KratosMultiphysics.kratos_utilities import CheckIfApplicationsAvailable
+
+# Importing the base class
+from KratosMultiphysics.CoSimulationApplication.solver_wrappers.kratos import kratos_base_wrapper
+
+# Importing StructuralMechanics
+if not CheckIfApplicationsAvailable("GeoMechanicsApplication"):
+    raise ImportError("The GeoMechanicsApplication is not available!")
+from KratosMultiphysics.GeoMechanicsApplication.geomechanics_analysis import GeoMechanicsAnalysis
+
+# Other imports
+from KratosMultiphysics.CoSimulationApplication.utilities import data_communicator_utilities
+
+def Create(settings, model, solver_name):
+    return GeoMechanicsWrapper(settings, model, solver_name)
+
+class GeoMechanicsWrapper(kratos_base_wrapper.KratosBaseWrapper):
+    """This class is the interface to the StructuralMechanicsApplication of Kratos"""
+
+    def _CreateAnalysisStage(self):
+        return GeoMechanicsAnalysis(self.model, self.project_parameters)
+
+    def _GetDataCommunicator(self):
+        if not KM.IsDistributedRun():
+            return KM.ParallelEnvironment.GetDataCommunicator("Serial")
+
+        # now we know that Kratos runs in MPI
+        parallel_type = self.project_parameters["problem_data"]["parallel_type"].GetString()
+
+        # first check if the solver uses MPI
+        if parallel_type != "MPI":
+            return data_communicator_utilities.GetRankZeroDataCommunicator()
+
+        # now we know that the solver uses MPI, only question left is whether to use all ranks or a subset
+        self._CheckDataCommunicatorIsConsistentlyDefined(self.project_parameters["solver_settings"]["model_import_settings"], self.settings["mpi_settings"])
+
+        return super()._GetDataCommunicator()
```

Comment:
I think this is not correct/handled better by the baseclass. Have you tested this/had problems?

Reply:
Hi @philbucher 
Thank you for your review. It is highly appreciated.
Can you please elaborate in more details what is the issue in this script? For consistency, I tried to built it as similar as possible to the scripts for other applications. 
The code runs and converges. We are trying to validate the results, however this is still on progress.
Do I need to apply changes in this script?

Reply:
hey, this is only about MPI. Are you using the CoSimApp in MPI? This particular part of the code is used to enable solvers to use less ranks than `MPI_COMM_WORLD` has.

I would like to understand what you need differently from e.g. the fluid or structural wrapper

Reply:
Hi @philbucher 
Thank you for your explanation.
We are not using CoSimApp with MPI, as the MPI is still not ready in the GeoMechanics application. So, I paid less attention to this part of the script.

However, it is preferly to also fix this part in order to make parts correct and ready, in the case we fix/add the MPI in GeoMechApp in the future. 

It is still not clear to me what we need differently from the other opplications, but as far as I can imagine, we will use the standard MPI, like the most of the other applications.